.Rprofile

```diff
@@ -1,45 +1,22 @@
-# source global .Rprofile (very important to load user specific settings)
-# DO NOT EDIT THIS LINE!
-if(file.exists("~/.Rprofile")) source("~/.Rprofile")
-
-# This profile can be used to link the model to a specified library snapshot
-# (e.g. if your model version is from an older date and does not work with the
-# newest libraries anymore). By default it is not active.
-
-# Gather all library snapshots
-snapshots <- sort(list.files(
-             path = '/p/projects/rd3mod/R/libraries/snapshots',
-             pattern = '^20[0-9]{2}_[0-9]{2}(_[0-9]{2})?$',
-             full.names = TRUE), decreasing = TRUE)
-latest_snapshot <- ifelse (length(snapshots) > 0, snapshots[[1]], NA)
-
-# Just uncomment the following line and set the snapshot path
-# to a path of your choice or use the second line to use the latest snapshot.
-# Please make also sure that in your config file this .Rprofile file is copied
-# to the model output folder. Otherwise, the run itself will again use the
-# default library set!
-
-# snapshot <- "/p/projects/rd3mod/R/libraries/snapshots/2022_03"
-# snapshot <- latest_snapshot
-
-if(exists("snapshot") && file.exists(snapshot)) {
-  message("libPaths was manually set to: ",snapshot)
-  .libPaths(snapshot)
-}
```

Comment:
These lines should definitely stay in some way. I use both the option to fix to a specific snapshot or to use the latest to guarantee reproducibility and also to avoid errors in longer runs if somebody breaks something in between

Reply:
`renv` supersedes snapshots.  
Each REMIND directory has its own private R package library, that is constant and does not see unsupervised changes (pending changes discussed above).  Each REMIND output directory gets an independent copy of that library.
So "fixing to a specific snapshot" becomes just "not changing the `renv` package library".

Reply:
Ok, thanks, this wasn't clear from the PR description. I also regularly set different REMIND repos to the same library snapshot for testing, so it would be great if this could easily be achieved.

Reply:
And, sorry for asking as I haven't understood the renv approach: In NGFS, we ran REMIND with a certain set of libraries, and later had to rerun the reporting with a more recent library version. Will there still be an easy way to get older versions back once the libraries were updated within the repo?

Reply:
> I also regularly set different REMIND repos to the same library snapshot for testing, so it would be great if this could easily be achieved.

Should suffice to copy the `renv.lock` file from one directory to the other to make the target behave like the source.

Reply:
> In NGFS, we ran REMIND with a certain set of libraries, and later had to rerun the reporting with a more recent library version. Will there still be an easy way to get older versions back once the libraries were updated within the repo?

See [here](https://rstudio.github.io/renv/articles/renv.html#workflow).  You basically work with different versions of the `renv.lock` file, however you choose to do that (`git`, `copy/rename`, …).

Reply:
Every `renv` is represented by a simple text file called `renv.lock`. In order to get the exact same `renv` in different places you could just copy `renv.lock` and then `renv::restore` it. In the same way you can save an `renv.lock` file anywhere, maybe in a `renv_locks` folder in your home dir, and re-use them later by copying and restoring.

Reply:
Thanks, Pascal. We have [this quite PIK cluster centered explanation](https://github.com/remindmodel/remind/blob/develop/tutorials/4_RunningREMINDandMAgPIE.md#create-snapshot-of-r-libraries) in the REMIND-MAgPIE-tutorial, which would then require an update, I guess.

Reply:
Thanks, I'll make sure it gets an update.